### PR TITLE
additional linked fixes

### PR DESCRIPTION
**Pull request by cmastr**

### CODE REVIEW

File: src/engage/audiences/linked-audiences.md [hunk 1]
```diff
@@ -43,15 +43,41 @@ This triggers a compute for the audience (where the audience conditions run on y
...
 
-| Event Conditions     | Description                           |
+| Conditions     | Description                           |
 |---------------------------|---------------------------------------|
 | Associated with an entity   | Creates a condition that filters profiles associated with entity relationships defined in the [Data Graph](/docs/unify/linked-profiles/data-graph/). With this condition, you can traverse the full nested entity relationship and filter your audience on entity column values. Note: you can only create nested entity conditions up to four levels in depth. For example, an entity condition that queries for relationships between Profiles, Accounts, Credit Cards, and Transactions has four levels of depth.       |
 | Where profile trait     | Creates a condition that filters profiles with a specific trait. |
 | Part of an audience     | Creates a condition that filters profiles that are part of an existing linked or classic audience. |
 | Performed event         | Creates a condition that filters profiles on their event history. Users can also filter on event property values.|
 
+#### Error States
```
Comment (cmastr) [inline src/engage/audiences/linked-audiences.md:55]: Error states for review.

File: src/engage/audiences/linked-audiences.md [hunk 2]
```diff
@@ -41,17 +41,43 @@ After creating the audience, you'll be redirected to the Overview page. By defau
...
+| SQL Compilation execution | Examples: extract phase failed: An unexpected Snowflake error occurred trying to execute your model query: 000904 (42000): SQL compilation error: error line 30 at position 61 invalid identifier 'EV.PROPERTIES'  extract phase failed: An unexpected Snowflake error occurred trying to execute your model query: 000904 (42000): SQL compilation error: error line 23 at position 19 invalid identifier 'UPDATES.TRAITS' |  |
+| Duplicate records for ID | Examples: extract phase failed: Duplicate records have been detected. This could be because the model query returned multiple records for the same ID column (MATCHID), or the Segment records table itself has duplicate records. |  |
+| Check results | An unknown error occurred. Execution will retry. If the problem persists, please contact Segment support. | Retry |
+| Extract results | Execution is paused due to an identified operational issue. | Pause for manual intervention |
+| Process results | Execution is paused due to an identified operational issue. | Set retl_results_processed = true on success  Set retl_results_processed = false on failure  Set lock  Abort other execution attempts  Page Compute |
```
Comment (smccoy-twilio) [inline src/engage/audiences/linked-audiences.md:78]: I think we should consolidate line 77 and 78 to be:
>| Process results | Execution is paused due to an identified operational issue. | Pause for manual intervention

reasons being:
- I think `extract results` is really covered by the warehouse errors
- I don't think we should expose the notes on line 78 (which are details re: how we implemented the pause internally)

File: src/engage/audiences/linked-audiences.md [hunk 3]
```diff
@@ -41,17 +41,43 @@ After creating the audience, you'll be redirected to the Overview page. By defau
...
+
+| Failure | External message | Workflow implication |
+|---|---|---|
+| Parse error | An unknown error occurred. Execution will retry. If the problem persists, please contact Segment support. | Terminate and execute next run |
+| Graph missing entity | Entity missing from Data Graph: <entity> | Terminate and execute next run |
+| Graph missing column for entity | Entity column missing from Data Graph for Entity <entity> and Column: <column> | Terminate and execute next run |
+| No columns exist for entity | No columns exist for entity in Data Graph: <entity> | Terminate and execute next run |
+| Missing join key between entities | Join key missing from Data Graph for relation: <relationship> | Terminate and execute next run |
+| Join key is wrong type | Relationship key should be a string: <relationship> | Terminate and execute next run |
+| Entity relationship not in Data Graph (we know the entity) | Entity relationship not in Data Graph for entity <entity> | Terminate and execute next run |
```
Comment (smccoy-twilio) [inline src/engage/audiences/linked-audiences.md:67]: maybe we could remove the `(we know the *` and instead have examples like you did with SQL Compilation errors. thinking the phrasing might be confusing as is.

File: src/engage/audiences/linked-audiences.md [hunk 4]
```diff
@@ -41,17 +41,43 @@ After creating the audience, you'll be redirected to the Overview page. By defau
...
+| Query fails for unknown internal error | An unexpected error occurred. Execution will retry. If the problem persists, please contact Segment support. | Retry |
+| Update model fails | An unexpected error occurred. Execution will retry. If the problem persists, please contact Segment support. | Retry |
+| Kick off rETL job | An unexpected error occurred. Execution will retry. If the problem persists, please contact Segment support. | Retry |
+| Warehouse errors | Pass through warehouse errors | Terminate and execute next run |
+| SQL Compilation execution | Examples: extract phase failed: An unexpected Snowflake error occurred trying to execute your model query: 000904 (42000): SQL compilation error: error line 30 at position 61 invalid identifier 'EV.PROPERTIES'  extract phase failed: An unexpected Snowflake error occurred trying to execute your model query: 000904 (42000): SQL compilation error: error line 23 at position 19 invalid identifier 'UPDATES.TRAITS' |  |
```
Comment (smccoy-twilio) [inline src/engage/audiences/linked-audiences.md:74]: not sure if you want to use a more generic example for `EV.PROPERTIES` and `UPDATES.TRAITS`

File: src/engage/audiences/linked-audiences.md [hunk 5]
```diff
@@ -41,17 +41,43 @@ After creating the audience, you'll be redirected to the Overview page. By defau
...
+| Duplicate records for ID | Examples: extract phase failed: Duplicate records have been detected. This could be because the model query returned multiple records for the same ID column (MATCHID), or the Segment records table itself has duplicate records. |  |
+| Check results | An unknown error occurred. Execution will retry. If the problem persists, please contact Segment support. | Retry |
```
Comment (smccoy-twilio) [inline src/engage/audiences/linked-audiences.md:76]: this is now `An unexpected error occurred attempting to execute this model query.`

File: src/engage/audiences/linked-audiences.md (hunk 3 above)
Comment (cmastr) [reply src/engage/audiences/linked-audiences.md:67]: Shannon can you provide the examples?

File: src/engage/audiences/linked-audiences.md (hunk 4 above)
Comment (cmastr) [reply src/engage/audiences/linked-audiences.md:74]: DONE

File: src/engage/audiences/linked-audiences.md (hunk 5 above)
Comment (cmastr) [reply src/engage/audiences/linked-audiences.md:76]: I assume 'this' is the Check Results error.

File: src/engage/audiences/linked-audiences.md (hunk 2 above)
Comment (cmastr) [reply src/engage/audiences/linked-audiences.md:78]: DONE

File: src/engage/audiences/linked-audiences.md (hunk 5 above)
Comment (smccoy-twilio) [reply src/engage/audiences/linked-audiences.md]: yep, that's right

File: src/engage/audiences/linked-audiences.md (hunk 3 above)
Comment (smccoy-twilio) [reply src/engage/audiences/linked-audiences.md]: sure thing. I think we can call this category of errors something like "Data Graph configuration errors".

examples would be:

- No columns exist for entity in Data Graph: 'account'
- Join key missing from Data Graph for relation: 'order_shipped'
- Relationship key should be a string: 123
- Entity relationship not in Data Graph for entity: 'account' and relationship: 'order_shipped'
- Entity relationship not in Data Graph: 'accounts_by_order'

Comment (smccoy-twilio) [reply src/engage/audiences/linked-audiences.md]: maybe we could also add that these are actionable by the user and once they update their configs to have everything set up correctly, subsequent executions should succeed past this step.

**Review by pwseg — APPROVED** [review]

**Pull request merged**